avb/libavb/uefi_avb_ops.c

```diff
@@ -271,7 +279,11 @@ static AvbIOResult write_rollback_index(__attribute__((unused)) AvbOps* ops,
     ret = EFI_SUCCESS;
   else {
 #if defined(SECURE_STORAGE_EFIVAR)
-    ret = write_efi_rollback_index(rollback_index_slot, rollback_index);
+#ifdef USE_TPM
+    if (!is_platform_secure_boot_enabled() ||
+        (ret = write_tpm2_rollback_index(rollback_index_slot, rollback_index)) == EFI_NOT_FOUND)
+#endif // USE_TPM
+      ret = write_efi_rollback_index(rollback_index_slot, rollback_index);
 #elif defined(SECURE_STORAGE_RPMB)
     ret = write_rpmb_rollback_index(rollback_index_slot, rollback_index);
```

Comment:
The use of compilation flag is impossible to follow.

Reply:
There is a auto detection solution at before and do not use condition compile flag. But there is not such requirement now.
If need it, then can add the support of it.

Reply:
I am always  in favor of automatic detection as you probably seen in my contribution to this project..